server/controllers/all.js

```diff
@@ -143,13 +176,65 @@ module.exports.export = async function(req, res) {
     }
 };
 
+module.exports.export = async function(req, res) {
+    try {
+        let passphrase = null;
+
+        if (req.body.encrypted) {
+            if (typeof req.body.passphrase !== 'string') {
+                throw new KError('missing parameter "passphrase"', 400);
+            }
+
+            passphrase = req.body.passphrase;
+            // Check password strength
+            if (!PASSPHRASE_VALIDATION_REGEXP.test(passphrase)) {
+                throw new KError('submitted passphrase is too weak', 400);
+            }
+        }
+
+        let ret = await getAllData();
+        ret.accesses = await Access.all();
+        // Only save user password if encryption is enabled.
+        ret = cleanData(ret, !!passphrase);
+        ret = JSON.stringify(ret, null, '   ');
+
+        if (passphrase) {
+            ret = encryptData(ret, passphrase);
+            res.setHeader('Content-Type', 'text/plain');
+        } else {
+            res.setHeader('Content-Type', 'application/json');
+        }
+
+        res.status(200).send(ret);
+    } catch (err) {
+        err.code = ERR_MSG_LOADING_ALL;
+        return asyncErr(res, err, 'when exporting data');
+    }
+};
+
 module.exports.import = async function(req, res) {
     try {
         if (!req.body.all) {
             throw new KError('missing parameter "all" in the file', 400);
         }
 
-        let world = req.body.all;
+        let world;
+        if (req.body.encrypted) {
+            if (typeof req.body.passphrase !== 'string') {
+                throw new KError('missing parameter "passphrase"', 400);
+            }
+
+            world = decryptData(req.body.all, req.body.passphrase);
+        } else {
+            world = new Buffer(req.body.all);
+        }
+
+        try {
```

Comment:
If I understand correctly, this has to be done only in the case where we had to decrypt the buffer: can you move this `try/catch` there, please? In this case, you can even remove the `world = new Buffer(...)` above.